go/teams/service_helper.go

```diff
@@ -122,14 +122,40 @@ func userVersionToDetails(ctx context.Context, g *libkb.GlobalContext, uv keybas
 }
 
 func userVersionsToDetails(ctx context.Context, g *libkb.GlobalContext, uvs []keybase1.UserVersion, forceRepoll bool) (ret []keybase1.TeamMemberDetails, err error) {
-	for _, uv := range uvs {
-		det, err := userVersionToDetails(ctx, g, uv, forceRepoll)
-		if err != nil {
-			return nil, err
+	ret = make([]keybase1.TeamMemberDetails, len(uvs), len(uvs))
+	type res struct {
+		i   int
+		det keybase1.TeamMemberDetails
+		err error
+	}
+	retCh := make(chan res, len(uvs))
+	width := 10
+	gate := make(chan struct{}, width)
+	for i, uv := range uvs {
+		go func(i int, uv keybase1.UserVersion) {
+
+			// This will block when there are more than `width` number of goroutines
+			// outstanding, and will unblock as soon as there is room.
+			gate <- struct{}{}
```

Comment:
list.go used `semaphore` for this: https://github.com/keybase/client/blob/master/go/teams/list.go#L82-L87

do we prefer one more than the other or it doesn't matter?

Reply:
i remember that, this seemed easier though, or does it seem too hacky?

Reply:
I'd say looks exactly how one would expect go code to look like :)

Reply:
Can actually swap 135 and 139

Reply:
Anyone else have opinions? I think mine is easier to understand but it doesn’t really matter.